src/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php

```diff
@@ -430,9 +430,22 @@ public function processMultiLineArray($phpcsFile, $stackPtr, $arrayStart, $array
                 }
 
                 if ($keyUsed === true && $tokens[$lastToken]['code'] === T_COMMA) {
-                    $error = 'No key specified for array entry; first entry specifies key';
-                    $phpcsFile->addError($error, $nextToken, 'NoKeySpecified');
-                    return;
+                    $nextToken = $phpcsFile->findNext(Tokens::$emptyTokens, ($lastToken + 1), null, true);
+
+                    /*
+                        Allow array unpacking
+
+                        $x = [
+                          'foo' => 'bar',
+                          ...$baz,
+                        ];
+                    */
+
```

Comment:
```suggestion
                    // Allow for PHP 7.4 array unpacking with an array declaration.
```

I don't think the code sample is needed here.